### ISSUE TITLE
Add support for Instagram

### ISSUE THREAD

**Issue by iandees** [osmlab/checkin-checker#8]
Instgram allows you to geotag a media posting with a location and they have support for webhooks. Let's add support for them.

**Comment by stevevance** [comment]
Yes! I geotag photos on Instagram more often than I check in on Foursquare.

**Comment by iandees** [comment]
A small hiccup here is that unlike Foursquare the Instagram webhooks don't send the user's e-mail address, so I'm going to have to build some database to link an Instagram user ID with an e-mail address that we ask for when the user connects.

**Comment by talllguy** [comment]
One issue is the IG locations are Facebook locations, which tend to be much lower quality (in my experience) unless they are claimed.

**Comment by stevevance** [comment]
Instagram locations also pull up massive boundaries, like "all of Chicago" or "all of [this neighborhood]". Does that pose a problem?

**Comment by dphiffer** [comment]
:+1:
